onnxruntime/core/framework/tunable.h

```diff
@@ -215,9 +215,24 @@ class TunableOp {
     }
   }
 
-  static double Profile(Op<ParamsT>& op, const ParamsT* param) {
+  static double Profile(Op<ParamsT>& op, const ParamsT* param, double min_time, bool early_stop) {
     constexpr const int num_iter = 100;
+    constexpr const int early_stop_num_iter = 10;
+    constexpr const double early_stop_factor = 1.2;
     TimerT timer{param->Stream()};
+
+    if (early_stop) {
+      timer.Start();
+      for (int i = 0; i < early_stop_num_iter; i++) {
+        ORT_THROW_IF_ERROR(op(param));
+      }
+      timer.End();
+      double cur_time = timer.Duration() / early_stop_num_iter;
+      if (cur_time > min_time * early_stop_factor) {
```

Comment:
> if (cur_time > min_time * early_stop_factor) {

What about the case where we get unlucky and the op instances are provided in descending order of their execution time? Then this early stop mechanism will not be very useful.

One idea to mitigate this might be to randomly shuffle the op instances outside of the `Profile` method.

Reply:
We may encounter this case, but the probability is small. And our instance id is the only instance information that we can use in debug process. I think we can keep the original order of instances id to facilitate debug. what's your opinion?